src/App.js

```diff
@@ -1,20 +1,441 @@
+      );
+    }
+  };
+
+  // Transfer tasks from In Progress to In Review
+  moveTaskInProgress = (id) => {
+    const taskToMove = this.state.tasksInProgress.find(
+      (task) => task.id === id
+    );
+    if (taskToMove) {
+      const newId = this.state.tasksInReview.length + 1;
+      const updatedTask = { ...taskToMove, id: newId };
+      this.setState(
+        (prevState) => ({
+          tasksInProgress: prevState.tasksInProgress.filter(
+            (task) => task.id !== id
+          ),
+          tasksInReview: [...prevState.tasksInReview, updatedTask],
+        }),
+        this.updateLocalStorage
+      );
+    }
+  };
+
+  // OPEN TASKS SECTION
+  updateTask = (id, updatedTask) => {
+    this.setState((prevState) => {
+      const updatedTasks = prevState.tasks.map((task) => {
+        if (task.id === id) {
+          return { ...task, ...updatedTask };
+        }
+        return task;
+      });
+      return { tasks: updatedTasks };
+    }, this.updateLocalStorage);
+  };
+
+  addTaskToDo = (task) => {
+    const newId = this.state.tasks.length;
+    const newTask = { ...task, id: newId };
+    this.setState(
+      (prevState) => ({ tasks: [...prevState.tasks, newTask] }),
+      this.updateLocalStorage
+    );
+  };
+
+  deleteTask = (id) => {
+    this.setState(
+      (prevState) => ({
+        tasks: prevState.tasks.filter((task) => task.id !== id),
+      }),
+      this.updateLocalStorage
+    );
+  };
+
+  updateLocalStorage = () => {
+    localStorage.setItem("tasks", JSON.stringify(this.state.tasks));
+    localStorage.setItem(
+      "tasksInProgress",
+      JSON.stringify(this.state.tasksInProgress)
+    );
+    localStorage.setItem(
+      "tasksInReview",
+      JSON.stringify(this.state.tasksInReview)
+    );
+    localStorage.setItem(
+      "tasksCompleted",
+      JSON.stringify(this.state.tasksCompleted)
+    );
+  };
+
+  resetLocalStorage = () => {
+    localStorage.clear();
+    this.setState(
+      {
+        tasks: [
+          {
+            id: 0,
+            title: "Reverse Engineer the Type-33 Needler",
+            task: "Coding a software to reverse engineer the Needler for Sacred Promissory",
+          },
+        ],
+      },
+      this.updateLocalStorage
+    );
+  };
+
+  componentDidMount() {
+    /** USED WHEN BROWSER IS REFRESHED OR REVISITED -> PERSISTENCE OF TASKS ACROSS SESSIONS
+     * componentDidMount() is used to retrieve data from 'localStorage' when the component, App.js, is mounted for the first time or refreshed in the browser
+     */
+    const savedTasks = localStorage.getItem("tasks");
+    const savedTasksInProgress = localStorage.getItem("tasksInProgress");
+    const savedTasksInReview = localStorage.getItem("tasksInReview");
+    const savedTasksCompleted = localStorage.getItem("tasksCompleted");
+
+    if (savedTasks) {
+      this.setState({
+        tasks: JSON.parse(savedTasks),
+        tasksInProgress: JSON.parse(savedTasksInProgress) || [],
```

Comment:
why use a default here, but not for the other states? You possibly set everything to undefined here